lib/hipchat/errors.rb

```diff
@@ -1,14 +1,49 @@
 module HipChat
-  class ServiceError        < StandardError; end
-  class UnknownRoom         < ServiceError; end
-  class RoomNameTooLong     < ServiceError; end
-  class RoomMissingOwnerUserId < ServiceError; end
-  class Unauthorized        < ServiceError; end
-  class UsernameTooLong     < ServiceError; end
-  class UnknownResponseCode < ServiceError; end
-  class UnknownUser         < ServiceError; end
-  class InvalidApiVersion   < ServiceError; end
-  class InvalidUrl          < ServiceError; end
-  class InvalidEvent        < ServiceError; end
-  class UnknownWebhook      < ServiceError; end
+  class ServiceError            < StandardError;       end
+  class RoomNameTooLong         < ServiceError;        end
+  class RoomMissingOwnerUserId  < ServiceError;        end
+  class UnknownResponseCode     < ServiceError;        end
+  class Unauthorized            < UnknownResponseCode; end
+  class BadRequest              < ServiceError;        end
+  class MethodNotAllowed        < ServiceError;        end
+  class UsernameTooLong         < ServiceError;        end
+  class InvalidApiVersion       < ServiceError;        end
+  class InvalidUrl              < ServiceError;        end
+  class InvalidEvent            < ServiceError;        end
+  class ObjectNotFound          < UnknownResponseCode; end
+  class UnknownRoom             < ObjectNotFound;      end
+  class UnknownUser             < ObjectNotFound;      end
+  class UnknownWebhook          < ObjectNotFound;      end
+  class TooManyRequests         < UnknownResponseCode; end
+
+  class ErrorHandler
+
+    # Pass-through to catch error states and raise their appropriate exceptions
+    # @param klass {Symbol} The class of object we are handling the error for
+    # @param identifier {String} An identifying string for the object causing the error
+    # @param response {HTTParty::Response} The HTTParty response/request object
+    def self.response_code_to_exception_for(klass, identifier, response)
```

Comment:
Using `klass` in this context is actually a ruby convention.

http://stackoverflow.com/questions/4299289/what-is-the-difference-between-class-and-klass-in-ruby